JsonLd/Serializer/ItemNormalizer.php

```diff
@@ -152,7 +152,7 @@ public function normalize($object, $format = null, array $context = [])
                 $subResource = $this->resourceResolver->getResourceFromType($collectionType)
             ) {
                 $values = [];
-                foreach ($attributeValue as $index => $obj) {
+                foreach ($attributeValue->getValues() as $index => $obj) {
```

Comment:
The getValues method is from Collection interface of symfony. So every collection on symfony have it. It prevent the issue that array collection internal array can be not ordered properly.

If you want we can add : 
```php
$values = $attributeValue instanceof Collection ? $attributeValue->getValues() : $attributeValue;
```